openhtf/__init__.py

```diff
@@ -71,21 +72,33 @@ def PhaseTwo(test):
       # Analyze widget integration status
 
     Test(PhaseOne, PhaseTwo).Execute()
+
+  Note that Test() objects *must* be created in the main thread, but can be
+  .Execute()'d in a separate thread.
   """
 
   def __init__(self, *phases, **metadata):
     code_info = test_record.CodeInfo.ForModuleFromStack(levels_up=2)
+    self._test_options = TestOptions()
     self._test_info = TestData(phases, metadata=metadata, code_info=code_info)
-    self._output_callbacks = []
-    self.loop = False
+    self._lock = threading.Lock()
+    self._stopped = False
+    signal.signal(signal.SIGINT, self.Stop)
```

Comment:
This sets the signal handler and ignores the previous one. If this was in `SetupFramework()`, then you could call that from the main thread then do your test creation/execution wherever you wanted